### ISSUE TITLE
[v0.23.0-rc5] authWithOAuth2 causes error on resubscribe

### ISSUE THREAD

**Issue by kaboc** [pocketbase/pocketbase#5674]
```dart
pb.collection('my_table').subscribe(
      '*',
      (event) { ... },
      filter: pb.filter('user_id = {:userId}', {'userId': userId}),
    );
```

Calling `authWithOAuth2()` after a subscription is started like above throws the following error:

> ClientException: {url: https://example.com/api/realtime, isAbort: false, statusCode: 403, response: {data: {}, message: The current and the previous request authorization don't match., status: 403}, originalError: null}

It does not seem to happen if the `filter` argument is omitted. I didn't see the error before v0.23.0-rcX.

### Steps to reproduce

1. Call `authWithOAuth2()` to log in/sign up.
2. Start a subscription with filtering.
3. Call `authWithOAuth2()` to link the user to another external account.
4. An error is thrown.

### Environment

- PocketBase v0.23.0-rc5 (used as a Go framework)
- Dart SDK for PocketBase 0.19.0-rc2
- Go 1.23.2
- Dart 3.4.4

**Comment by kaboc** [comment]
I also tried specifying a record ID as the topic. It resulted in the same error.

```dart
pb.collection('my_table').subscribe('record id', (event) { ... });
```

As an aside, it looks like the callback is not called if a particular record is watched in this way. I'm not sure if it is an issue specific to the current RC version since it was the first time I had tried passing an ID to the method.

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it and I don't think it is v0.23.0 specific.

The error message indicates that you already had realtime subscription while being authenticated with another account and then after the OAuth2 you attempt to subscribe again but the SDK reuses the already existing realtime connection which cause an error because the auth state has changed, aka. the following operations could have caused it:

- auth with user1
- realtime subscribe (start realtime connection with user1)
- auth with user2
- realtime subscribe (because the authStore has changed, this will be with user2)
- fails because you can't replace the realtime connection user once established 

One possible solution is to call pb.realtime.unsubsribe() before auth.

I've added it in my todo to eventually consider doing automatically as part of the JS and Dart SDK auth flow, but for now remains a low priority.

**Comment by kaboc** [comment]
@ganigeorgiev 
Thank you for explaining. I don't think the user has changed because it happens while I'm still trying to obtain an auth URL. It means I've just started an auth flow and haven't switched to a new user yet. The user ID hasn't changed either at that point. It didn't happen in previous versions, so I suspected it was a bug introduced in one of the newer versions.

> One possible solution is to call pb.realtime.unsubsribe() before auth.

I think it'll work as a tentative workaround. Thank you.
(I hope the root cause of the issue will be addressed sometime in the future.)

**Comment by ganigeorgiev** [comment]
Again, this is an expected behavior.

As the error message indicates changing the auth user of a realtime connection is not supported at the moment.

Keep in mind that even if you restart your app, depending on how the SDK was initialized and whether you use persistent store, the `pb.authStore` may not be empty if you were previously testing with another user before sending the `authWithOAuth2` call.

**Comment by kaboc** [comment]
I have no idea where in my steps the user can switch. I made sure to delete existing users (from both the server and the client) and clear the auth store before restarting the app. However, there may be something wrong in my custom auth store. I'll look into it. It is probably a problem unique to me if no one else has not experienced the same issue. Thank you again.

**Comment by ganigeorgiev** [comment]
Actually there is a problem and I was able to reproduce it (it is caused by the v-23.0 apis and it happens when there is a follow up subscription from the same user because we are not extracting properly the current auth user id and it compares the old user id against an empty string).

I'll push an update a little bit later today.

**Comment by kaboc** [comment]
Oh, great! Thank you so much.

**Comment by ganigeorgiev** [comment]
I've pused a fix in [v0.23.0-rc6](https://github.com/pocketbase/pocketbase/releases/tag/v0.23.0-rc6).

Note that the scenario with the 2 different users reusing the same auth connection shown in https://github.com/pocketbase/pocketbase/issues/5674#issuecomment-2411335454 is still not supported.

**Comment by kaboc** [comment]
The error is gone now. Thanks a lot!
